src/templates/mongoose/templateHelpers.ts

```diff
@@ -0,0 +1,24 @@
+import { mongoosePrimitiveTypes } from '../../model/dataTypes/primitiveDataTypes';
+import DataType from "../../model/dataTypes/DataType";
+
+export function getMongoosePrimitive(type: DataType): string {
+  // TODO: this is a workaround for the typeName holding a normalized name instead of an original name
+  let typeName = type.typeName.replace("Primitive", "").toLowerCase();
+  return mongoosePrimitiveTypes[typeName] || typeName;
+}
+
+const SCHEMA_FUNCTION_REQUIRED = new Set(['DomainResource', 'Resource', 'BackboneElement', 'Element', 'Quantity', 'ElementDefinition']);
```

Comment:
Codacy found an issue: [Variable name SCHEMA_FUNCTION_REQUIRED must match one of the following formats: camelCase](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5738534)